checkov/secrets/plugins/custom_regex_detector.py

```diff
@@ -216,3 +216,12 @@ def analyze_string(self, string: str, **kwargs: Optional[Dict[str, Any]]) -> Gen
                         yield submatch, regex
                 else:
                     yield match, regex
+
+
+def find_line_number(file_string: str, substring: str, default_line_number: int) -> int:
```

Comment:
this goes over the entire file which can be big
instead we can start from the line number and stop when the string contains the secret